### PR TITLE
Complete the unit testing and add some features

### PR DESCRIPTION
**Pull request by peter279k**
## First of all, thank you for your great package. Let me access OpenWeatherMap API with PHP easily.
# Change log
- Complete the unit testing.

- Ignore testing the ```getWeatherHistory``` method in ```OpenWeatherMap.php```.
This method is ignored because the api key need to have a paid permission.

- Integrate the ```integTests```  with ```tests```.

- Replace ```sprintf``` function with ```str_replace``` in ```Weather.php``` because the character ```%s``` is proper for using the ```sprintf```.

- Replace ```json_last_error_msg``` with ```json_last_error``` in ```OpenWeatherMap.php```. Because the ```json_last_error_msg``` is available for  PHP 5.5+ (see this [link](http://php.net/manual/en/function.json-last-error-msg.php)). In order to be compatible with older PHP version (like 5.3 or 5.4), so we do this.

- In ```ExampleCache.php```, adding ```setTempPath``` method allow us to specify the cached file path.

- Using the ```#``` to comment message is deprecated in ``*.ini`` file. Replace the ```#``` with this ```;```.

-  Set the proper configuration in ```phpunit.xml.dist``` file.

- Add the required extension: ```ext-curl``` in composer.json.
It will check whether the ```curl``` extension is installed during running the ```Composer``` installation.

- Add the ```require-dev``` key in ```composer.json```.

the [travis-ci result](https://travis-ci.org/peter279k/OpenWeatherMap-PHP-Api)
the [testing code coverage report](https://codecov.io/gh/peter279k/OpenWeatherMap-PHP-Api) (provided by ```codecov```)

### CODE REVIEW

**Comment by peter279k** [comment]
Hi @cmfcmf , I fix the icoUrl variable error.

File: Cmfcmf/OpenWeatherMap/WeatherHistory.php [hunk 1]
```diff
@@ -75,6 +75,9 @@ public function __construct($weatherHistory, $query)
 
         $utctz = new \DateTimeZone('UTC');
         foreach ($weatherHistory['list'] as $history) {
+            if (isset($history['city'])) {
+                continue;
+            }
```
Comment (cmfcmf) [inline Cmfcmf/OpenWeatherMap/WeatherHistory.php]: Why did you add this?

Comment (peter279k) [inline Cmfcmf/OpenWeatherMap/WeatherHistory.php]: Hi @cmfcmf , I didn't add this.

**Comment by cmfcmf** [comment]
This has been merged as part of #96. Thank you @peter279k!
